ASoC: SOF: topology: fix incorrect widget id in error message

yongzhi1 opened the pull request:
In sof_widget_ready(), use swidget->comp_id to print the intended value in message "error: failed to add widget id %d ..".

yongzhi1 committed:
ASoC: SOF: topology: remove incorrect widget id in error message

In sof_widget_ready() function, the shift field of
struct snd_soc_tplg_dapm_widget is incorrectly used to print
widget id in dev_err(scomp->dev, "error: failed to add widget id %d ..",
this patch removes the useless tw->shift from the error output.

Signed-off-by: Yong Zhi <yong.zhi@intel.com>

ranj063 commented on sound/soc/sof/topology.c:
```diff
@@ -1532,7 +1532,7 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		dev_err(scomp->dev,
 			"error: failed to add widget id %d type %d name : %s stream %s\n",
-			tw->shift, swidget->id, tw->name,
+			swidget->comp_id, swidget->id, tw->name,
```
@yongzhi1 is the comp_id even useful for anything? Why not remove this altogether?

yongzhi1 replied:
Ack, let's remove it :)

ranj063 commented on sound/soc/sof/topology.c:
```diff
@@ -1531,10 +1531,9 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
-			tw->shift, swidget->id, tw->name,
-			strnlen(tw->sname, SNDRV_CTL_ELEM_ID_NAME_MAXLEN) > 0
-				? tw->sname : "none");
+			"error: failed to add widget type %d name : %s stream %s\n",
```
while we're at it, lets also remove the error prefix in the message and also in the commit message can you mention that the widget id is incorrect and thats why its not useful

plbossart commented on sound/soc/sof/topology.c:
```diff
@@ -1531,10 +1531,9 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
-				? tw->sname : "none");
+			"error: failed to add widget type %d name : %s stream %s\n",
+			swidget->id, tw->name, strnlen(tw->sname, SNDRV_CTL_ELEM_ID_NAME_MAXLEN) > 0
+							? tw->sname : "none");
```
The commit message doesn't really describe the changes.

The initial code used tw->shift for the id, but now sdwidget->id is shown as a widget type?

it's likely yet another of those cases where people used 'id' for different things...

yongzhi1 replied:
Ack, to use tw->shift for widget id is a mistake in the current code, so this patch removes it.
swidget->id is defined as DAPM widget type, should I change the line to "failed to add widget of DAPM type %d .."?

yongzhi1 commented:
V3:
1. updated commit message
2. removed "error:" in the print.

Old message:
```
sof-audio-pci-intel-mtl 0000:00:1f.3: error: failed to add widget id 0 type 28 name : dai-copier.DMIC.dmic01.capture stream dmic01
```
With the patch:
```
sof-audio-pci-intel-mtl 0000:00:1f.3: failed to add widget type 28 name : dai-copier.DMIC.dmic01.capture stream dmic01
```
